content/posts/2019/05/first-contribution.eng.md

```diff
@@ -0,0 +1,70 @@
+---
+title: First Contribution to Libplanet
+date: 2019-05-07
+authors: [seunghun.lee]
+translators: [kidon.seo]
+---
+
+
+Hello, I'm Seung Hoon Lee, member of Planetarium Dev Team. I believe one of the reasons why working at Planetarium is so attractive is that I can work as an open source developer. This piece is about my first contribution to Project [Libplanet][]. For an introduction to Libplanet, please see 〈[Libplanet 0.2 릴리스][1]〉
+
+[Libplanet]: https://github.com/planetarium/libplanet
+[1]: {{< ref "libplanet-0.2.eng.md" >}}
+
+Why did I contribute to Libplanet?
+----------------------------
+
+In January, Planetarium announced its first recruiting notice and revealed a repository for Project Libplanet. Although I thought it was an attractive project by an impressive team, I was hesitant to apply because Libplanet was being developed in C# to work with Unity (my main expertise was Python). After a brief hesitation, however, I made up my mind to participate in Libplanet anyways because it was an open source project. I decided to participate in basic issues at first and see how it went.
+ 
+C# Development Environment 
+------------
+
+As mentioned earlier, I used to mainly develop in Python on Mac. I often contributed to Python open source projects that interested me and this was easy because most Python developers also developed on Mac and were familiar with the open source environment. But as I was about to build a C# development environment on Mac, I found myself stuck from the very first step, unsure about what to install. Fortunately, there were projects like [Mono][] and [.NET Core][2] that enabled .NET development on Mac, and editors such as [VS Code][] that also supported plug-ins for C# development. So I was able to set up an environment without much difficulty. Now that Libplanet also provides a [developer guide][3] and a [environment setup guide][4], I think it will be much easier  for developers who want to contribute to the project.
+
+[Mono]: https://www.mono-project.com/
+[2]: https://en.wikipedia.org/wiki/.NET_Core
+[VS Code]: https://code.visualstudio.com/
+[3]: https://github.com/planetarium/libplanet/blob/master/CONTRIBUTING.md
+[4]: https://gist.github.com/dahlia/5333634f62509293cd46c0e4ba65b2f5
+
+What to Contribute?
+------------------
+
+Libplanet’s address uses [Ethereum's][5] address format, which uses a checksum in a combination of upper and lower case letters proposed in [EIP-55][]. For example:
+
+Hexadecimal Method
+:  `0xd1220a0cf47c7b9be7a2e6ba89f429762e7b9adb`
+
+EIP-55 Checksum
+:  `0xD1220A0cf47c7B9Be7A2E6BA89F429762e7b9aDb`
+
+
+When I first contributed to Libplanet, the project used a hexadecimal method consisting of only lowercase letters for its address format. Therefore, this issue had to be implemented so that the address string representation was expressed in the EIP-55 checksum format used by Ethereum. For more information, see the corresponding [issue][6] and [PR][7].
```

Comment:
```suggestion
When I first contributed to Libplanet, the project used a hexadecimal consisting of only lowercase letters for its address format. Therefore, this issue had to be implemented so that the address string representation was expressed in the EIP-55 checksum format used by Ethereum. For more information, see the corresponding [issue][6] and [pull request][7].
```